### ISSUE TITLE
Unary Operators in Fuzzilli differ from the JS definition

### ISSUE THREAD

**Issue by WilliamParks** [googleprojectzero/fuzzilli#101]
In working on #4, I ran into what appears to be a discrepancy in how Fuzzilli definies a unary operator (++, --), implemented [here](https://github.com/googleprojectzero/fuzzilli/blob/master/Sources/Fuzzilli/FuzzIL/Operations.swift#L456) vs what I understand the JS definition to be (+, 1), described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_negation).

**Comment by saelo** [comment]
Hi! Can you elaborate a bit more on what exactly the issue is that you are running into?
Is it that the `++` and `--` operators in FuzzIL are actually translated to `x + 1` and `x - 1` respectively or that the negation operator is missing (this should probably just be added...)?
For the `++` and `--` operators, it should be possible to translate something like 
```
let i = 42;
i++;
```

into 
```
v0 = LoadInt 42
v1 = Phi v0
v2 = ++ v1
Copy v1 v2
```

But I understand that that's probably not very convenient/intuitive. I'm open for discussions about how to implement these operators "properly" in FuzzIL. The main problem is probably that they could only operate on Phis as they modify their input?

**Comment by WilliamParks** [comment]
I think the best resolution to this issue would be implementing "standard" unary +/-.

Mainly, I was ran into this when the parser I'm using reported `+v0` and `-v0`, as unary plus/minus, which fuzzilli then turned into `v0 + 1` and `v0 - 1` respectively!

**Comment by saelo** [comment]
Ah yeah, I agree, that should be fixable by adding `+` and `-` as unary operators. Feel free to send a PR for that :)

**Comment by WilliamParks** [comment]
Closed by #114
